### ISSUE TITLE
11.0.0-beta.0 - Private Instance Methods - Parsing error: params is not iterable

### ISSUE THREAD

**Issue by filmic** [babel/babel-eslint#749]
I'm getting following error:
`Parsing error: params is not iterable`

while linting a file with a private instance method.

Using:
`@babel/core@7.2.2`
`@babel/plugin-proposal-class-properties@7.3.0`
`@babel/plugin-proposal-private-methods@7.3.2`
`babel-eslint@11.0.0-beta.0`
`eslint@5.13.0`

`test.js` (linted file):
```
class Person {
  #age = 19;

  #increaseAge() {
    this.#age++;
  }

  birthday() {
    this.#increaseAge();
    alert("Happy Birthday!");
  }
}
```

`.eslintrc.js`:
```
module.exports = {
	"parser": "babel-eslint",
};
```

`babel.config.js`:
```
module.exports = {
	presets: [],
	plugins: [
		"@babel/plugin-proposal-class-properties",
		"@babel/plugin-proposal-private-methods"
	]
};
```

**Comment by hadnet** [comment]
Same thing here.

**Comment by hadnet** [comment]
I resolved that problem adding in .eslintrc `"extends": [
        "plugin:eslint-plugin/classPrivateMethods"
    ]`

and using this versions:
"babel-eslint": "^10.0.1",
"eslint": "^5.13.0"

**Comment by satazor** [comment]
I'm also experiencing the same issue, I got around it temporarily by using arrow functions instead:

```js
 #increaseAge = () => {
    this.#age++;
  }
```

**Comment by satazor** [comment]
Actually, with the arrow function syntax I get:

> Unexpected 'this'. (babel/no-invalid-this)

**Comment by satazor** [comment]
After digging, I found this related issue: https://github.com/babel/babel/issues/9506

**Comment by filmic** [comment]
The issue in Babel (babel/babel#9506) is blocked by estree/estree#180.

**Comment by IAmJulianAcosta** [comment]
I can confirm this:

```
    "@babel/plugin-proposal-class-properties": "^7.3.4",
    "@babel/plugin-proposal-private-methods": "^7.3.4",
    "eslint": "5.15.1",
    "babel-eslint": "11.0.0-beta.0"
```

**Comment by sepehr** [comment]
I also second this:

```
"@babel/plugin-proposal-class-properties": "^7.5.5",
"@babel/plugin-proposal-private-methods": "^7.6.0",
"eslint": "6.1.0",
"babel-eslint": "11.0.0-beta.0"
```

**Comment by kaicataldo** [comment]
This is caused by https://github.com/babel/babel/issues/9506.

**Comment by 1nfility** [comment]
Any fix..?

**Comment by filmic** [comment]
It has been fixed in https://github.com/babel/babel/pull/10914 (Babel v7.8.0)
